### PR TITLE
Add minimal architecture definition for Aarch64

### PR DESCRIPTION
**Pull request by Keno**
The first commit re-arranges some of our ptrace support to avoid trying to access `Arch::` data structures that don't exist on Aarch64 because these ptrace commands are not support. The remaining two commits add a skeleton implementation of the AArch64 arch definition. `Registers` support is missing, but there's some trickiness there, so that'll be a separate PR.

### CODE REVIEW

File: src/kernel_abi.h [hunk 1]
```diff
@@ -1969,6 +1972,85 @@ struct X86Arch : public BaseArch<SupportedArch::x86, WordSize32Defs> {
...
+    blksize_t st_blksize;
+    int __pad2;
+    blkcnt_t st_blocks;
+    struct timespec st_atim;
+    struct timespec st_mtim;
+    struct timespec st_ctim;
+    int __rr_unused[2];
+  };
+
+  struct stat64 {
```
Comment (khuey) [inline src/kernel_abi.h:2002]: AArch64 doesn't have a stat64 afaict. So why is this here?

File: src/record_syscall.cc [hunk 2]
```diff
@@ -2556,6 +2577,20 @@ static Switchable prepare_ptrace(RecordTask* t,
...
+    case PTRACE_POKEDATA:
+    case PTRACE_PEEKUSER:
+    case PTRACE_POKEUSER:
+    case PTRACE_GETREGS:
+    case PTRACE_GETFPREGS:
+    case PTRACE_GETFPXREGS:
+    case PTRACE_SETREGS:
+    case PTRACE_SETFPREGS:
+    case PTRACE_SETFPXREGS:
+      prepare_ptrace_legacy<Arch>(t, syscall_state);
```
Comment (khuey) [inline src/record_syscall.cc:2592]: I didn't check very hard that you copy/pasted the code here correctly :)

File: src/kernel_abi.h (hunk 1 above)
Comment (Keno) [reply src/kernel_abi.h:2002]: syscalls.py uses fstatat64 for all architectures. There should probably be a separate entry for fstatat that uses regular struct stat, then this can be dropped. Will try that.

Comment (Keno) [reply src/kernel_abi.h:2002]: Actually, even if we do that, it'll still fail to build because the code generated from syscalls.py is not stratified by architecture. Doing that seems a bit overkill. Should we `typedef sts64 stat` instead on architectures that don't have the stat64 syscall?

Comment (Keno) [reply src/kernel_abi.h:2002]: Actually a typedef doesn't work because `Arch::stat64` is already the syscall, but we could do
```
struct stat64 : public stat {};
```
for 64 bit archs.

Comment (khuey) [reply src/kernel_abi.h:2002]: Ok, yeah, now that I look at it I see x86_64 has the same thing going on. We could clean that up in a followup with the inheritance you suggest but no need to block this.

File: src/Monkeypatcher.cc [hunk 3]
```diff
@@ -930,6 +944,17 @@ void patch_at_preload_init_arch<X64Arch>(RecordTask* t,
                                         params.syscall_patch_hooks);
 }
 
+template <>
+void patch_at_preload_init_arch<AA64Arch>(RecordTask* t,
+                                         Monkeypatcher&) {
+  auto params = t->read_mem(
+      remote_ptr<rrcall_init_preload_params<X64Arch>>(t->regs().arg1()));
```
[the remark this answers is not in the document]
Comment (Keno) [reply src/Monkeypatcher.cc]: Sorry, snuck in while splitting this into a sensibly sized commit ;). Fixed.

File: src/GdbConnection.h [hunk 4]
```diff
@@ -581,6 +581,7 @@ class GdbConnection {
   enum {
     CPU_64BIT = 0x1,
```
[the remark this answers is not in the document]
Comment (Keno) [reply src/GdbConnection.h]: Done

File: src/record_syscall.cc (hunk 2 above)
Comment (Keno) [reply src/record_syscall.cc:2592]: Well, I didn't touch it, but hopefully the tests would catch it if I had ;). I did catch #2559 while doing this copy pasting so hopefully it's actually better than before :).

File: src/kernel_abi.h (hunk 1 above)
Comment (Keno) [reply src/kernel_abi.h]: I included the inheritance thing here. Seems to work fine.
